tests/generated_artifacts/test_generated_traces.py

```diff
@@ -0,0 +1,32 @@
+from pathlib import Path
+
+import pytest
+from any_agent import AgentTrace
+
+ASSETS_PATH = Path(__file__).parent.parent / "assets"
+
+
+@pytest.fixture(
+    params=list(ASSETS_PATH.rglob("*_trace.json")),
+    ids=lambda x: x.parent.name,
+)
+def generated_trace(request: pytest.FixtureRequest) -> AgentTrace:
+    return AgentTrace.model_validate_json(request.param.read_text())
+
+
+def test_any_tool_used(generated_trace: AgentTrace):
+    assert any(span.is_tool_execution() for span in generated_trace.spans), "No tools were used"
+
+
+@pytest.mark.parametrize("max_steps", range(5, 30, 5))
+def test_steps_taken(generated_trace: AgentTrace, max_steps: int):
```

Comment:
These are just a couple of examples on how to implement a test with different thresholds and combine it with `xfail`. See #88 of an example using this to verify improvements